packages/service-worker/src/services/settings/handlers/wallet_setSettings.ts

```diff
@@ -0,0 +1,192 @@
+import {
+  DAppRequestHandler,
+  DAppProviderRequest,
+  JsonRpcRequestParams,
+  Languages,
+  CURRENCIES,
+  AnalyticsConsent,
+} from '@core/types';
+import { injectable } from 'tsyringe';
+import { SettingsService } from '../SettingsService';
+import { z } from 'zod';
+import { ethErrors } from 'eth-rpc-errors';
+
+const SettingsSchema = z.object({
+  language: z.nativeEnum(Languages).optional(),
+  currency: z.nativeEnum(CURRENCIES).optional(),
+  analyticsConsent: z.nativeEnum(AnalyticsConsent).optional(),
+  theme: z.enum(['LIGHT', 'DARK', 'SYSTEM']).optional(),
+  preferredView: z.enum(['floating', 'sidebar']).optional(),
+  showTokensWithoutBalances: z.boolean().optional(),
+  coreAssistant: z.boolean().optional(),
+  showTrendingTokens: z.boolean().optional(),
+  tokensVisibility: z
+    .record(z.string(), z.record(z.string(), z.boolean()))
+    .optional(),
+  collectiblesVisibility: z
+    .record(z.string(), z.record(z.string(), z.boolean()))
+    .optional(),
+});
+
+type PartialSettings = z.infer<typeof SettingsSchema>;
+type Params = [settings?: PartialSettings];
+
+type Currency = 'USD' | 'EUR' | 'GBP' | 'AUD' | 'CAD' | 'CHF' | 'HKD';
+export interface WalletSetSettingsResponse {
+  currency: Currency;
```

Comment:
We should use the same types as in the schema in Zod for both handlers.
Right now adding a new language or currency would mean we need to update multiple enums in multiple unrelated files.